src/services/safari-view-controller.ts

```diff
@@ -0,0 +1,25 @@
+import { SafariViewController } from '@ionic-native/safari-view-controller'
+import { InAppBrowser } from '@ionic-native/in-app-browser'
+
+const openWithNativeBrowser = async (url: string) => {
+    InAppBrowser.create(url, '_system')
+}
+
+export const openWithSafariViewController = async (url: string) => {
+    const isAvailable = await SafariViewController.isAvailable()
+    if (isAvailable) {
+        SafariViewController.show({
+            url,
+            hidden: false,
+            animated: false,
+            transition: 'curl',
+            enterReaderModeIfAvailable: true,
+            tintColor: '#ff0000',
+        }).subscribe(
+            (result: any) => console.log(result.event),
+            (error: any) => console.error(error)
```

Comment:
maybe fallback to openWithNativeBrowser on error?

Reply:
Since this is an observable, the `subscribe` method could be passed several errors during the lifespan of the `safari-view-controller` window, which means we would open several native browser windows in this case